jsonrpc/websocket.go

```diff
@@ -0,0 +1,168 @@
+package jsonrpc
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"net/http"
+	"time"
+
+	"github.com/NethermindEth/juno/utils"
+	"nhooyr.io/websocket"
+)
+
+type Websocket struct {
+	rpc        *Server
+	log        utils.SimpleLogger
+	connParams *WebsocketConnParams
+	port       uint16
+}
+
+func NewWebsocket(port uint16, methods []Method, log utils.SimpleLogger) *Websocket {
+	ws := &Websocket{
+		rpc:        NewServer(),
```

Comment:
Maybe in the next PR we could make HTTP and WS share the same RPC server

Reply:
I agree that we can reduce a lot of code duplication, but I think we will want separate servers since subscribe endpoints can only be served with websockets.